src/recorder.js

```diff
@@ -39,39 +39,24 @@ export default class Recorder {
 
       // Reset this state.
       this.recorder = null;
-      this.isRecording = false;
 
       options.onStop?.({ url, media, mimeType, dimensions });
     };
-
-    this.isRecording = false;
-    this.isPaused = false;
   }
 
   start() {
-    if (!this.isRecording) {
-      this.recorder.start();
-      this.isRecording = true;
-    }
+    this.recorder.start();
   }
 
   pause() {
-    if (!this.isPaused) {
-      this.recorder.pause();
-      this.isPaused = true;
-    }
+    this.recorder.pause();
   }
 
   resume() {
-    if (this.isPaused) {
-      this.recorder.resume();
-      this.isPaused = false;
-    }
+    this.recorder.resume();
   }
 
   stop() {
-    if (this.isRecording) {
-      this.recorder.stop();
-    }
+    this.recorder.stop();
```

Comment:
So this call is idempotent? 

Commit message could use an explanation why removing this is ok. A couple of other commits as well.

Reply:
My thinking here was this: These checks that were there before were not consistent, IIRC, and I saw little value in making them so. I think it's the responsibility of the caller to check the state beforehand, and the caller in this case already manages its own version of these variables, which is the right thing to do.

To answer your actual question: No, it's not idempotent. You get an error when you call these control methods in the wrong state.